packages/form-js-viewer/src/render/components/form-fields/Number.js

```diff
@@ -55,94 +54,120 @@ export function Numberfield(props) {
 
   const inputRef = useRef();
 
-  const [ stringValueCache, setStringValueCache ] = useState('');
+  const [ cachedValue, setCachedValue ] = useState(value);
+  const [ displayValue, setDisplayValue ] = useState(value);
 
-  // checks whether the value currently in the form data is practically different from the one in the input field cache
-  // this allows us to guarantee the field always displays valid form data, but without auto-simplifying values like 1.000 to 1
-  const cacheValueMatchesState = useMemo(() => Numberfield.config.sanitizeValue({ value, formField: field }) === Numberfield.config.sanitizeValue({ value: stringValueCache, formField: field }), [ stringValueCache, value, field ]);
+  const sanitize = useCallback((value) => Numberfield.config.sanitizeValue({ value, formField: field }), [ field ]);
 
-  const displayValue = useMemo(() => {
+  const [ debouncedOnChange, flushOnChange ] = useFlushDebounce(props.onChange);
 
-    if (value === 'NaN') return 'NaN';
-    if (stringValueCache === '-') return '-';
-    return cacheValueMatchesState ? stringValueCache : ((value || value === 0) ? Big(value).toFixed() : '');
+  const previousCachedValue = usePrevious(value);
 
-  }, [ stringValueCache, value, cacheValueMatchesState ]);
-
-  const arrowIncrementValue = useMemo(() => {
-
-    if (incrementValue) return Big(incrementValue);
-    if (decimalDigits) return Big(`1e-${decimalDigits}`);
-    return Big('1');
+  useEffect(() => {
```

Comment:
I think you can take out the `useEffect` here and just have the if statement running in the render.